test/e2e/storage/pv_reclaimpolicy.go

```diff
@@ -91,6 +92,62 @@ var _ = SIGDescribe("PersistentVolumes [Feature:ReclaimPolicy]", func() {
 		})
 
 		/*
+			Test Steps:
+			1. Create vmdk
+			2. Create PV Spec with volume path set to VMDK file created in Step-1, and PersistentVolumeReclaimPolicy is set to Delete
+			3. Create PVC with the storage request set to PV's storage capacity.
+			4. Wait for PV and PVC to bound.
+			5. Delete PVC.
+			6. Verify volume is attached to the node and volume is accessible in the pod.
+			7. Verify PV status should be failed.
+			8. Delete the pod.
+			9. Verify PV should be detached from the node and automatically deleted.
+		*/
+		It("should not detach and unmount PV when associated pvc with delete as reclaimPolicy is deleted when it is in use by the pod", func() {
+			vsp, err := vsphere.GetVSphere()
+			Expect(err).NotTo(HaveOccurred())
+
+			volumePath, pv, pvc, err = testSetupVSpherePersistentVolumeReclaim(vsp, c, ns, v1.PersistentVolumeReclaimDelete)
+			Expect(err).NotTo(HaveOccurred())
+			// Wait for PV and PVC to Bind
+			framework.ExpectNoError(framework.WaitOnPVandPVC(c, ns, pv, pvc))
+
+			By("Creating the Pod")
+			pod, err := framework.CreateClientPod(c, ns, pvc)
+			Expect(err).NotTo(HaveOccurred())
+			node := types.NodeName(pod.Spec.NodeName)
+
+			By("Deleting the Claim")
+			framework.ExpectNoError(framework.DeletePersistentVolumeClaim(c, pvc.Name, ns), "Failed to delete PVC ", pvc.Name)
+			pvc = nil
+
+			By("Verify the volume is attached to the node")
+			isVolumeAttached, verifyDiskAttachedError := verifyVSphereDiskAttached(vsp, pv.Spec.VsphereVolume.VolumePath, node)
+			Expect(verifyDiskAttachedError).NotTo(HaveOccurred())
+			Expect(isVolumeAttached).To(BeTrue())
+
+			By("Verify the volume is accessible and available in the pod")
+			verifyVSphereVolumesAccessible(pod, []*v1.PersistentVolume{pv}, vsp)
+			framework.Logf("Verified that Volume is accessible in the POD after deleting PV claim")
+
+			// Verify PV is Present, after PVC is deleted and PV status should be Failed.
```

Comment:
Maybe PV verification should come before vsphere volume verification.  That way we ensure that PV controller processed the API update.

Then again, that still doesn't guarantee that A/D controller or kubelet saw the update.

Reply:
@msau42 Moved logic to wait until PV phase becomes `v1.VolumeFailed` before checking for volume attachment and volume accessibility.